src/Core/SonarLintNotifications/INotificationSettingStorage.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SonarLint.VisualStudio.Core.SonarLintNotifications
+{
+    public enum StorageLocation
+    {
+        User,
+        VSInstance,
+        Solution
+    }
+
+    public interface INotificationSettingStorage
+    {
+        void SaveNotificationSetting(INotificationSetting setting, StorageLocation storageLocation);
```

Comment:
I'm not sure if Setting is the right word here. It sounds like a configuration.

How about a simpler API, only for dismissed notifications?

`void SaveBlockedNotification(string notificationId)`
`IEnumerable<string> GetBlockedNotifications()`